phonenumbers.go

```diff
@@ -3327,7 +3328,8 @@ func GetTimezonesForPrefix(number string) ([]string, error) {
 	// strip any leading +
 	number = strings.TrimLeft(number, "+")
 
-	for i := timezoneMap.MaxLength; i > 0; i-- {
+	matchLength := int(math.Min(float64(timezoneMap.MaxLength), float64(len(number))))
+	for i := matchLength; i > 0; i-- {
```

Comment:
I think nicer to avoid the casting and just make this an if..

```go
matchLength := len(number)
if matchLength > timezoneMap.MaxLength {
    matchLength = timezoneMap.MaxLength
}